src/python/PSetTweaks/WMTweak.py

```diff
@@ -435,16 +433,16 @@ def makeJobTweak(job):
     logging.info("Adding %d files to 'fileNames' attr", len(primaryFiles))
     logging.info("Adding %d files to 'secondaryFileNames' attr", len(secondaryFiles))
     if len(primaryFiles) > 0:
-        result.addParameter("process.source.fileNames", primaryFiles)
+        result.addParameter("process.source.fileNames", "customTypeCms.untracked.vstring(%s)" % primaryFiles)
         if len(secondaryFiles) > 0:
```

Comment:
This code - that was there even before your changes - is somehow surprising to me. I think this `if` statement for secondary files should be at the same upper level as the one for `primaryFiles`. Otherwise, what happens to a job without primary files but with pileup files? Or is it a case that is impossible to happen?